### ISSUE TITLE
Tracking progress

### ISSUE THREAD

**Issue by saphal1998** [XifeiNi/jslings#12]
There are two facets to tracking progress here - 

1. Overall category
2. Within a particular category

To approach this, I propose the following strategy - 

1. Add a `.userdata` file in the directory structure that keeps tracks of completed tests. The status of the tests can be 'pending', 'success' and 'failure'. Every new test needs to be added to this file with the status of `pending`.

2. Build a test gen script to add tests, a token doc could look like
```ts
<random unique identifier>
// I am not done - Or some other decorator

...insert code
```
The reasoning behind adding a unique identifier is simply to keep track of tests completed

We can hence immediately fetch the progress of the user by simply querying this file.

- Other nits
1. Also need to add a reset command that simply erases the user's progress. The benefit of this, is we simply need to delete the `userdata` file to reset the status of the user.

Let's discuss and iterate over this idea

**Comment by amorriscode** [comment]
I'm curious how this is different from `exercises.json`? To me that's already where we track the progress of the tests.

Second, I like the test generator idea. I created #13 for capturing discussion on that.

**Comment by saphal1998** [comment]
As discussed on call (writing this just to keep everyone up to date), `exercises.json` can contain the information of the different tests available on the CLI and `.userdata` merely tracks user progress.

This bifurcation is useful in the case a user wants to reuse `jslings` in which case the user can simply delete `.userdata` and start with a clean slate

**Comment by amorriscode** [comment]
@saphal1998 moving the `exercises.json`/`.userdata` discussion here so we don't pollute the PR.

If we are tracking `status` in _both_ files, how do cleanly avoid corruption? For example, a user manually resets progress by deleting `.userdata`. The `exercises.json` file is now completely out of sync with the project. What then?

**Comment by saphal1998** [comment]
Okay, I think I may have missed the mark with my communication.

- `exercises.json` - The file is important for the CLI to work. It captures two things
    1) General Information about the exercises, such as testPath, concept etc. 
    2) The sequence of execution of tests
- '.userdata' simply contains the names/ids of the exercises the user has completed. Hence this is user-specific and tracks user progress.

So, I don't really think there is an overlap of data and we don't have to worry about syncing the information between the two.
Does this make sense?

**Comment by amorriscode** [comment]
Closed by #17.
